src/components/Header/Header.jsx

```diff
@@ -8,10 +8,10 @@ import SearchComponent from '../search/SearchComponent'
 import client from '../../utils/client';
 import { useNavigate } from 'react-router-dom';
 
-const Header = ({ role, setSearchInput }) => {
+const Header = ({ setSearchInput }) => {
   let navigate = useNavigate();
 
-  console.log("This is the role", role)
+  const role = localStorage.getItem("role")
```

Comment:
I like the idea of getting the role from localstorage, however, I'm not sure if it's the right place to get it. The `App.js` component has a role state. That state gets passed down to a few different components. 

If you were to add this line into the `App.js` component, and use whatever role you get from the localstorage as the initial value of the role state, we could then pass the saved role down to all components that need it.

Reply:
I tried it this way but the header remained at the top of the login page?

Reply:
Although I dont know why it would make a difference? But that seemed to happen with me